### DISCUSSION TITLE
What this project is all about?

### DISCUSSION THREAD

**Discussion by WrongBit** [qwertie/ecsharp#119] [Q&A]
I read a whole readme and still have no any idea, what you wrote and why. I guess you're proud of your work, but all of that is just "one more useless repo", because you cannot explain WHAT you made. SuperC#?? Sorry, make no sense to me.

**Comment by qwertie** [comment]
Hi, please visit the [LeMP home page](http://ecsharp.net/lemp/) to understand what it is. Actually I just updated it with a new "compile-time code execution and code manipulation" example, but I waited 20 minutes and the changes are not showing up. You can see the changes [here instead](https://github.com/qwertie/ecsharp/blob/gh-pages/lemp/index.md).

As for _why_ I made it, it's hard to say... but LeMP/EC# is just a small part of my much larger dream to create an ecosystem of general-purpose tools for understanding and manipulate code from multiple programming languages (see [loyc.net](http://loyc.net/)). Progress has been very slow due to the huge size of the goal and lack of interest among other people (because I often find other people's desires motivating to me).

**Comment by WrongBit** [comment]
Hi, David! Thank you for reply! Later I found info, but problem is same info better to have here too - in ReadMe. Then people will easier understand what project is.
I found that most of EC# features already done by MS. Irony, huh! :) But what impress me you know about Nemerle! THIS is the project of future. Now it evolved into next gen as a "Nitra" - more powerful "constructor" of compilers - sure this topic will be interested to you. Good luck!
